pkg/config/engine/containerd/containerd.go

```diff
@@ -89,14 +91,16 @@ func New(opts ...Option) (engine.Interface, error) {
 		return (*ConfigV1)(cfg), nil
 	case 2:
 		return cfg, nil
+	case 3:
+		return (*ConfigV3)(cfg), nil
 	}
 
 	return nil, fmt.Errorf("unsupported config version: %v", version)
 }
 
 // parseVersion returns the version of the config
 func (c *Config) parseVersion(useLegacyConfig bool) (int, error) {
```

Comment:
We've essentially replaced the `useLegacyConfig` variable with the `configVersion` option. Would it make sense to update the implementation here to something like:

```
defaultVersion := 3
if c.configVersion != 0 {
    defaultVersion = c.configVersion
}
```
and drop the `useLegacyConfig` argument?